src/ng/compile.js

```diff
@@ -2245,43 +2337,58 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
               directiveNormalize(nodeName), 'E', maxPriority, ignoreDirective);
 
           // iterate over the attributes
-          for (var attr, name, nName, ngAttrName, value, isNgAttr, nAttrs = node.attributes,
+          for (var attr, name, nName, ngAttrName, value, ngPrefixMatch, nAttrs = node.attributes,
                    j = 0, jj = nAttrs && nAttrs.length; j < jj; j++) {
             var attrStartName = false;
             var attrEndName = false;
 
+            var isNgAttr = false, isNgProp = false, isNgEvent = false;
+            var multiElementMatch;
+
             attr = nAttrs[j];
             name = attr.name;
             value = attr.value;
 
-            // support ngAttr attribute binding
+            // support ng-attr-*, ng-prop-* and ng-on-*
             ngAttrName = directiveNormalize(name);
-            isNgAttr = NG_ATTR_BINDING.test(ngAttrName);
-            if (isNgAttr) {
+            if (ngPrefixMatch = ngAttrName.match(NG_PREFIX_BINDING)) {
+              isNgAttr = ngPrefixMatch[1] === "Attr";
+              isNgProp = ngPrefixMatch[1] === "Prop";
+              isNgEvent = ngPrefixMatch[1] === "On";
+
               name = name.replace(PREFIX_REGEXP, '')
-                .substr(8).replace(/_(.)/g, function(match, letter) {
+                .substr(4 + ngPrefixMatch[1].length).replace(/_(.)/g, function(match, letter) {
                   return letter.toUpperCase();
                 });
-            }
 
-            var multiElementMatch = ngAttrName.match(MULTI_ELEMENT_DIR_RE);
-            if (multiElementMatch && directiveIsMultiElement(multiElementMatch[1])) {
+            // support *-start / *-end multi element directives
+            } else if ((multiElementMatch = ngAttrName.match(MULTI_ELEMENT_DIR_RE)) && directiveIsMultiElement(multiElementMatch[1])) {
               attrStartName = name;
               attrEndName = name.substr(0, name.length - 5) + 'end';
               name = name.substr(0, name.length - 6);
             }
 
             nName = directiveNormalize(name.toLowerCase());
             attrsMap[nName] = name;
-            if (isNgAttr || !attrs.hasOwnProperty(nName)) {
-                attrs[nName] = value;
-                if (getBooleanAttrName(node, nName)) {
-                  attrs[nName] = true; // presence means true
-                }
+
+            if (isNgProp) {
+              attrs[ngAttrName] = value;
```

Comment:
Is this correct? Which attribute should have the `value? If any?

Reply:
Is attrs the attrs object that gets passed to the link fn / controller? I don't think ngProp should set the attribute here.

Reply:
Seems right to me.

Reply:
Yes this `attrs` is what gets passed to the compile / link / controller.

Note that here we are assigning `attrs.ngPropFoo`, not `attrs.foo` like `ng-attr-foo` would. Also the value is the expression that will later get `$parsed`.  While I agree it's weird putting anything prop related on `attrs`, I did this so we can...
a) use the existing `createEventDirective` from ngEventDirs.js
b) follow the same pattern as ngEventDirs.js and `ng-attr-*` where the value that gets `$parse`-ed (or `$interpolate`-ed for `ng-attr`) is read of the attribute in the compile step of the directive

I'm not sure how to achieve both of those without putting the expression onto `attrs`...

Ideas? Opinions?